python/pyarrow/tests/test_convert_builtin.py

```diff
@@ -2000,6 +2000,17 @@ def test_roundtrip_nanosecond_resolution_pandas_temporal_objects():
         pd.Timestamp(9223371273709551616, unit='ns')
     ]
 
+    ty = pa.timestamp('ns', tz='US/Eastern')
+    value = 1604119893000000000
+    arr = pa.array([value], type=ty)
+    data = arr.to_pylist()
+    assert isinstance(data[0], pd.Timestamp)
+    restored = pa.array(data, type=ty)
+    assert arr.equals(restored)
+    assert restored.to_pylist() == [
+        pd.Timestamp(value, unit='ns', tz='US/Eastern')
```

Comment:
```suggestion
        pd.Timestamp(value, unit='ns').tz_localize("UTC").tz_convert('US/Eastern')
```

(it's the same, but I find it a bit more explicit, otherwise you need to know if pd.Timestamp interprets the integer as wall time or UTC time)